src/script/view_model/ContentViewModel.ts

```diff
@@ -158,84 +261,35 @@ export class ContentViewModel {
 
       if (!conversationEntity) {
-        rightSidebar.close();
-
-        throw new ConversationError(
-          ConversationError.TYPE.CONVERSATION_NOT_FOUND,
-          ConversationError.MESSAGE.CONVERSATION_NOT_FOUND,
-        );
+        return this.closeRightSidebarAndThrowError();
       }
 
       const isActiveConversation = this.conversationState.isActiveConversation(conversationEntity);
 
       if (!isActiveConversation) {
-        rightSidebar.close();
+        this.closeRightSidebar();
       }
 
-      const isConversationState = contentState === ContentState.CONVERSATION;
-      const isOpenedConversation = conversationEntity && isActiveConversation && isConversationState;
-
-      if (isOpenedConversation) {
-        if (openNotificationSettings) {
-          rightSidebar.goTo(PanelState.NOTIFICATIONS, {entity: conversationEntity});
-        }
-        return;
-      }
-
-      setContentState(ContentState.CONVERSATION);
-      this.mainViewModel.list.openConversations();
+      const isOpenedConversation = this.isConversationOpen(conversationEntity, isActiveConversation);
 
+      this.handleConversationState(isOpenedConversation, openNotificationSettings, conversationEntity);
       if (!isActiveConversation) {
         this.conversationState.activeConversation(conversationEntity);
       }
 
-      const messageEntity = openFirstSelfMention ? conversationEntity.getFirstUnreadSelfMention() : exposeMessageEntity;
-
-      if (conversationEntity.is_cleared()) {
-        conversationEntity.cleared_timestamp(0);
-      }
-
-      if (conversationEntity.is_archived()) {
-        await this.conversationRepository.unarchiveConversation(conversationEntity);
-      }
-
-      this.changeConversation(conversationEntity, messageEntity);
-      this.showContent(ContentState.CONVERSATION);
-      this.previousConversation = this.conversationState.activeConversation();
-      setHistoryParam(
-        generateConversationUrl({id: conversationEntity?.id ?? '', domain: conversationEntity?.domain ?? ''}),
-      );
-
-      if (openNotificationSettings) {
-        rightSidebar.goTo(PanelState.NOTIFICATIONS, {entity: this.conversationState.activeConversation() ?? null});
-      }
-    } catch (error) {
-      const isConversationNotFound = error.type === ConversationError.TYPE.CONVERSATION_NOT_FOUND;
-      if (isConversationNotFound) {
-        PrimaryModal.show(
-          PrimaryModal.type.ACKNOWLEDGE,
-          {
-            text: {
-              message: t('conversationNotFoundMessage'),
-              title: t('conversationNotFoundTitle', Config.getConfig().BRAND_NAME),
-            },
-          },
-          undefined,
-        );
+      this.processMessageEntity(conversationEntity, openFirstSelfMention, exposeMessageEntity);
+      this.showAndNavigate(conversationEntity, openNotificationSettings);
+    } catch (error: any) {
+      if (this.isConversationNotFoundError(error)) {
+        this.showErrorModal();
```

Comment:
The name is a little too generic, It's not showing any error modal, it's showing the conversation not found error modal. 
I think it was okay to inline that code (doesn't really need to be extracted to a function)

Reply:
I think its good to extract it, makes the catch block much more compact and readable. And we can reuse this conversation error modal in future. I can rename this to showConversationErrorModal. what do you think?

Reply:
Then let's make it very explicit what it's doing and maybe name it `showConversationNotFoundErrorModal` in this case.